odoo/addons/base/models/ir_qweb.py

```diff
@@ -295,45 +298,71 @@ def get_asset_bundle(self, xmlid, files, env=None):
         'xml' not in tools.config['dev_mode'],
         tools.ormcache_context('xmlid', 'options.get("lang", "en_US")', 'css', 'js', 'debug', 'async_load', 'defer_load', 'lazy_load', keys=("website_id",)),
     )
-    def _get_asset_nodes(self, xmlid, options, css=True, js=True, debug=False, async_load=False, defer_load=False, lazy_load=False, values=None):
-        files, remains = self._get_asset_content(xmlid, options, css, js)
+    def _get_asset_nodes(self, xmlid, options, css=True, js=True, debug=False, async_load=False, defer_load=False, lazy_load=False, media=None, values=None):
+        nodeAttrs = None
+        if css and media:
+            nodeAttrs = {
+                'media': media,
+            }
+        files, remains = self._get_asset_content(xmlid, options, nodeAttrs)
         asset = self.get_asset_bundle(xmlid, files, env=self.env)
+        remains = [node for node in remains if (css and node[0] == 'link') or (js and node[0] == 'script')]
         return remains + asset.to_node(css=css, js=js, debug=debug, async_load=async_load, defer_load=defer_load, lazy_load=lazy_load)
 
     def _get_asset_link_urls(self, xmlid, options):
         asset_nodes = self._get_asset_nodes(xmlid, options, js=False)
         return [node[1]['href'] for node in asset_nodes if node[0] == 'link']
 
-    # @tools.ormcache_context('xmlid', 'options.get("lang", "en_US")', keys=("website_id",))
-    def _get_asset_content(self, xmlid, options, css=True, js=True):
+    @tools.ormcache_context('xmlid', 'options.get("lang", "en_US")', keys=("website_id",))
+    def _get_asset_content(self, xmlid, options, nodeAttrs=None):
         options = dict(options,
             inherit_branding=False, inherit_branding_auto=False,
             edit_translations=False, translatable=False,
             rendering_bundle=True)
 
         options['website_id'] = self.env.context.get('website_id')
-        addons = module_boot()
-        addon_files = self.env['ir.asset'].get_addon_files(addons=addons, bundle=xmlid, css=css, js=js)
+
+        addons = []
+        if not request:
+            addons = self.env['ir.module.module']._installed_sorted()
+        else:
+            addons = module_boot()
+
+        addon_files = self.env['ir.asset'].get_addon_files(addons=addons, bundle=xmlid, css=True, js=True)
 
         files = []
         remains = []
         for _, file in addon_files:
+            ext = file.split('.')[-1]
+            is_js = ext in SCRIPT_EXTENSIONS
+            is_css = ext in STYLE_EXTENSIONS
+            if not is_js and not is_css:
```

Comment:
Same here: if `js=True` the `get_addon_files` method should only return JS paths/urls, and the same applies to `css=True`. If this is necessary I guess there is a problem to fix in `ir_asset.py` (or I'm just missing something).